packages/hardhat-vyper/src/index.ts

```diff
@@ -1,14 +1,25 @@
-import { TASK_COMPILE } from "hardhat/builtin-tasks/task-names";
-import { extendConfig, task } from "hardhat/internal/core/config/config-env";
+import {
+  TASK_COMPILE_GET_COMPILATION_TASKS,
+  TASK_COMPILE_SOLIDITY,
+} from "hardhat/builtin-tasks/task-names";
+import { extendConfig, subtask } from "hardhat/internal/core/config/config-env";
 
+import { TASK_COMPILE_VYPER } from "./task-names";
 import "./type-extensions";
 
 extendConfig((config) => {
   const defaultConfig = { version: "latest" };
   config.vyper = { ...defaultConfig, ...config.vyper };
 });
 
-task(TASK_COMPILE, async (_, { config, artifacts }) => {
+subtask(
+  TASK_COMPILE_GET_COMPILATION_TASKS,
+  async (): Promise<string[]> => {
```

Comment:
This is a little picky, but when you override a task you can call the original one, so here I'd do

```ts
  async (_, _, runSuper): Promise<string[]> => {
    const otherTasks = await runSuper();
    return [...otherTasks, TASK_COMPILE_VYPER];
  }
```

This way the plugin is more robust and can interact with others that also override this task.

Reply:
Exactly what I was looking for.